disk/datadog_checks/disk/disk.py

```diff
@@ -214,14 +213,19 @@ def _collect_inodes_metrics(self, mountpoint):
     def collect_latency_metrics(self):
         for disk_name, disk in psutil.disk_io_counters(True).iteritems():
             self.log.debug('IO Counters: {0} -> {1}'.format(disk_name, disk))
-            # x100 to have it as a percentage,
-            # /1000 as psutil returns the value in ms
-            read_time_pct = disk.read_time * 100.0 / 1000.0
-            write_time_pct = disk.write_time * 100.0 / 1000.0
-            self.rate(self.METRIC_DISK.format('read_time_pct'),
+            try:
+                # x100 to have it as a percentage,
+                # /1000 as psutil returns the value in ms
+                read_time_pct = disk.read_time * 100.0 / 1000.0
+                write_time_pct = disk.write_time * 100.0 / 1000.0
+                self.rate(self.METRIC_DISK.format('read_time_pct'),
                       read_time_pct, device_name=disk_name)
-            self.rate(self.METRIC_DISK.format('write_time_pct'),
+                self.rate(self.METRIC_DISK.format('write_time_pct'),
                       write_time_pct, device_name=disk_name)
+            except AttributeError:
+                # Some OS don't return read_time/write_time fields
+                # http://psutil.readthedocs.io/en/latest/#psutil.disk_io_counters
+                self.log.debug("Latency metrics not collected for {0}".format(disk_name))
```

Comment:
We should log why it's not being collected. Like: `Latency metrics not available for {0}" ?

Reply:
thanks @hush-hush. I added the error when logging to be more clear why it's not collected. let me know what you think